parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetInputFormat.java

```diff
@@ -147,6 +147,8 @@ public class ParquetInputFormat<T> extends FileInputFormat<Void, T> {
    */
   public static final String OFF_HEAP_DECRYPT_BUFFER_ENABLED = "parquet.decrypt.off-heap.buffer.enabled";
 
+  public static final String EAGERLY_READ_FULL_ROW_GROUP = "parquet.read.materializerowgroup";
```

Comment:
this could also be an Integer that controls how many pages to buffer at a time (-1 to fully materialize all pages)